modules/olm-enabling-operator-restricted-network.adoc

```diff
@@ -18,18 +19,128 @@ As an Operator author, your Operator must meet additional requirements to run pr
 
-For the CSV requirements, you can make the following changes as the Operator author.
+For the CSV requirements, you can make the following changes as the Operator author. The following procedure uses the Memcached Operator created in the Operator SDK tutorials for Go, Ansible, and Helm.
 
 .Prerequisites
 
 * An Operator project with a CSV.
 
 .Procedure
 
-. Use SHA references to related images in two places in the CSV for your Operator:
+. Add the images and tags used by the Operator by setting an environment variable, such as `RELATED_IMAGE_MEMCACHED`, in the `config/manager/manager.yaml` file:
++
+.Example `config/manager/manager.yaml`
+[%collapsible]
+====
+[source,yaml]
+----
+...
+spec:
+  ...
+    spec:
+      ...
+      containers:
+      - command:
+        - /manager
+        ...
+        env:
+        - name: RELATED_IMAGE_MEMCACHED
+          value: "docker.io/memcached@sha256:00b68b00139155817a8b1d69d74865563def06b3af1e6fc79ac541a1b2f6b961"
+----
+====
+
+. Replace hard-coded image references with environment variables in the following files:
+
+* For Go-based Operator projects, update the `controllers/memcached_controller.go` file with the following changes:
++
+.`controllers/memcached_controller.go` diff
+[%collapsible]
+====
+[source,diff]
+----
+...
+// deploymentForMemcached returns a memcached Deployment object
+...
+	Spec: corev1.PodSpec{
+        	Containers: []corev1.Container{{
+-			Image:   "memcached:1.4.36-alpine",
++			Image:   "docker.io/memcached@sha256:00b68b00139155817a8b1d69d74865563def06b3af1e6fc79ac541a1b2f6b961",
+			Name:    "memcached",
+			Command: []string{"memcached", "-m=64", "-o", "modern", "-v"},
+			Ports: []corev1.ContainerPort{{
+...
+----
+====
+
+* For Ansible-based Operator projects, update the `roles/memcached/tasks/main.yml` file with the following changes:
++
+.`roles/memcached/tasks/main.yml` diff
+[%collapsible]
+====
+[source,diff]
+----
+...
+spec:
+  containers:
+  - name: memcached
+    command:
+    - memcached
+    - -m=64
+    - -o
+    - modern
+    - -v
+-   image: "docker.io/memcached:1.4.36-alpine"
++   image: "{{ lookup('env', 'RELATED_IMAGE_MEMCACHED') }}"
+    ports:
+      - containerPort: 11211
+...
+----
+====
+
+* For Helm-based Operator projects, update the `helm-charts/memchached/values.yaml` file with the following changes:
++
+.`helm-charts/memchached/values.yaml` diff
+[%collapsible]
+====
+[source,diff]
+----
+  ## Memcached image and tag
+  ## ref: https://hub.docker.com/r/library/memcached/tags/
+  ##
+- image: memcached:1.5.20
++ image: "{{ lookup('env', 'RELATED_IMAGE_MEMCACHED') }}"
+...
+----
+====
+
+. Update the `Makefile` file with the following changes:
++
+.`Makefile` diff
+[source,diff]
+----
+...
+- $(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
++ $(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_GEN_FLAGS)
```

Comment:
```suggestion
+ $(KUSTOMIZE) build config/manifests | operator-sdk generate bundle $(BUNDLE_GEN_FLAGS)
$(BUNDLE_GEN_FLAGS)
```

You will also need to instruct users to add the definition of `BUNDLE_GEN_FLAGS` as well. In a newly generated `Makefile` this will be at the top with the other variable definitions, but just needs to be before this line:

```Makefile
BUNDLE_GEN_FLAGS ?= -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)

# USE_IMAGE_DIGESTS defines if images are resolved via tags or digests
# You can enable this value if you would like to use SHA Based Digests
# To enable set flag to true
USE_IMAGE_DIGESTS ?= false
ifeq ($(USE_IMAGE_DIGESTS), true)
	BUNDLE_GEN_FLAGS += --use-image-digests
endif
```